packaging/build_rpm.py

```diff
@@ -0,0 +1,375 @@
+                ))
+
+    rpms = {}
+    for f in listdir(RPMS_DIR):
+        f = f.strip()
+        if any(f.endswith(y) for y in ('.x86_64.rpm', '.noarch.rpm')):
+            name = check_output([
+                'rpm', '-qp', '--queryformat', '%{NAME}', f,
+                ])
+            rpms[name] = f
+
+    logger.info(('found RPMs:', rpms))
+
+    for dep in dependencies:
+        dep = dep.strip()
+        if '://' in dep:
+            package_name = check_output([
+                    'rpm', '-qp', '--queryformat', '%{NAME}', dep,
+                    ])
+            install_package = path_join(source, basename(dep))
+        else:
+            package_name = dep
+            install_package = rpms[dep]
+
+        check_call([MOCK, '--yum-cmd', 'remove', package_name])
+        check_call([
+            MOCK, '--yum-cmd', 'install', install_package,
+            ])
+
+
+def get_dependency_urls(spec_file):
+    dependencies = get_dependencies(spec_file)
+
+    return [dep.strip() for dep in dependencies if '://' in dep]
+
+
+def has_cmd(cmd):
+    try:
+        check_call(['which', cmd])
+        return True
+    except CalledProcessError:
+        return False
+
+
+def install_mock():
+    # EPEL is enabled only for the build system.
+    # It is not required on production systems.
+    run_vagrant('sudo yum -y install epel-release')
+    run_vagrant('sudo yum -y install mock')
+
+    # mock requires the user to be in the `mock` group
+    run_vagrant('sudo usermod -a -G mock $USER')
+
+
+def check_mock_config():
+    # allow network access during the build
+    # (we have to download packages from pypi)
+    mock_config = check_output([
+        MOCK, '--debug-config',
+        ]).splitlines()
+
+    for line in mock_config:
+        if all(s in line for s in ('rpmbuild_networking', 'True')):
+            break
+    else:
+        logger.info('configuring rpmbuild_networking')
+        check_call(
+                '''echo "config_opts['rpmbuild_networking'] = True" '''
+                '| sudo -n tee -a /etc/mock/site-defaults.cfg',
+                shell=True,
+                )
+
+
+def download_if_newer(url, outfile=None):
+    """Check the modified time and ask the server if the file has changed"""
+    if not outfile:
+        outfile = basename(url)
+    headers = {}
+    try:
+        mod_time = getmtime(outfile)
+    except OSError:
+        pass
+    else:
+        headers['If-Modified-Since'] = datetime.utcfromtimestamp(
+                mod_time).strftime('%a, %d %b %Y %H:%M:%S GMT')
+
+    req = urllib2.Request(url, headers=headers)
+    try:
+        try:
+            f = urllib2.urlopen(req)
+        except urllib2.HTTPError as f:
+            if f.getcode() == 304:
+                return
+            else:
+                raise
+        with open(outfile, 'wb') as out:
+            print('writing', outfile)
+            out.write(f.read())
+    finally:
+        f.close()
+
+
+def get_sources(spec_file):
```

Comment:
Could use a bit of a docstring here. Not exactly sure what are those sources we're getting

Reply:
Ok, I understand _what_ we're doing, but I'm not sure why. Why define something in a spec file if we're going to parse it like this then?

Reply:
Because defining `Source*:` in the spec file is the "right" way to do it. I'm probably twisting the definition of "right" by also dumping the repo tree in `SOURCES`, but if at some point we decide to ship proper `.src.rpm`s this will be one less thing to fix, at least